src/traccuracy/_tracking_graph.py

```diff
@@ -267,172 +254,51 @@ def get_location(self, node_id):
         """
         return [self.graph.nodes[node_id][key] for key in self.location_keys]
 
-    def get_nodes_with_flag(self, attr):
-        """Get all nodes with specified NodeAttr set to True.
+    def get_nodes_with_flag(self, flag: NodeFlag) -> list[Hashable]:
+        """Get all nodes with specified NodeFlag set to True.
 
         Args:
-            attr (traccuracy.NodeAttr): the node attribute to query for
+            flag (traccuracy.NodeFlag): the node flag to query for
 
         Returns:
-            (List(hashable)): A list of node_ids which have the given attribute
+            (List(hashable)): A list of node_ids which have the given flag
                 and the value is True.
         """
-        if not isinstance(attr, NodeAttr):
-            raise ValueError(f"Function takes NodeAttr arguments, not {type(attr)}.")
-        return list(self.nodes_by_flag[attr])
+        if not isinstance(flag, NodeFlag):
+            raise ValueError(f"Function takes NodeFlag arguments, not {type(flag)}.")
+        return list(self.nodes_by_flag[flag])
```

Comment:
I remember we had a conversation about casting the set to a list before returning here, but I can't remember why we thought it was necessary. What was the logic?

Reply:
@msschwartz21 I agree that I don't remembery why we casted to a list. I tried it as a set and all the tests pass. @DragaDoncila  I looked at the networkx code to see if using `NodeView` or `EdgeView` makes sense, and even within networkx they only use those specific classes for `.nodes` and `.edges`. Other functions on the graph either return an iterable of ids (e.g. `Graph.neighbors()` [here](https://github.com/networkx/networkx/blob/862b0e95d2008c7e032fbdb907168927c24b4255/networkx/classes/graph.py#L1315)) or a different, custom dict-like view (e.g. `AdjacencyView` is returned from `DiGraph.succ()` [here](https://github.com/networkx/networkx/blob/862b0e95d2008c7e032fbdb907168927c24b4255/networkx/classes/digraph.py#L384), `DegreeView` from `Graph.degree()` [here](https://github.com/networkx/networkx/blob/862b0e95d2008c7e032fbdb907168927c24b4255/networkx/classes/graph.py#L1482)). This makes me want to keep it as a set or an generic iterable of ids for the `nodes_by_flag` case. If we end up needing the dictionary like access to attributes we can look into using an `AdjacencyView` or one of the other views, but I think it might be unnecessarily complex if an iterator will do and is consistent enough.

Reply:
@cmalinmayor thanks for looking at what `nx` is doing. I think since they just return iterables I think it's fine to keep as a set - I'd hate to cast for no reason.